executor/infoschema_reader_test.go

```diff
@@ -149,6 +149,39 @@ func (s *testInfoschemaTableSuite) TestCharacterSetCollations(c *C) {
 		testkit.Rows("utf8mb4_bin utf8mb4"))
 }
 
+func (s *testInfoschemaTableSuite) TestDDLJobs(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("create database if not exists test_ddl_jobs")
+	tk.MustQuery("select db_name, job_type from information_schema.DDL_JOBS limit 1").Check(
+		testkit.Rows("test_ddl_jobs create schema"))
+
+	tk.MustExec("use test_ddl_jobs")
+	tk.MustExec("create table t (a int);")
+	tk.MustQuery("select db_name, table_name, job_type from information_schema.DDL_JOBS where table_name = 't'").Check(
+		testkit.Rows("test_ddl_jobs t create table"))
+
+	tk.MustQuery("select job_type from information_schema.DDL_JOBS group by job_type having job_type = 'create table'").Check(
+		testkit.Rows("create table"))
+
+	// Test the privilege of new user for information_schema.DDL_JOBS.
+	tk.MustExec("create user DDL_JOBS_tester")
+	DDLJobsTester := testkit.NewTestKit(c, s.store)
+	DDLJobsTester.MustExec("use information_schema")
+	c.Assert(DDLJobsTester.Se.Auth(&auth.UserIdentity{
+		Username: "DDL_JOBS_tester",
+		Hostname: "127.0.0.1",
+	}, nil, nil), IsTrue)
+	DDLJobsTester.MustQuery("select DB_NAME from information_schema.DDL_JOBS;").Check(
+		[][]interface{}{})
+
+	// Test the privilege of user with privilege of mysql for information_schema.ddl_jobs.
+	tk.MustExec("CREATE ROLE mysql_priv;")
+	tk.MustExec("GRANT ALL PRIVILEGES ON mysql.* TO mysql_priv;")
+	tk.MustExec("GRANT mysql_priv TO DDL_JOBS_tester;")
+	DDLJobsTester.MustExec("set role mysql_priv")
+	DDLJobsTester.MustQuery("select DISTINCT DB_NAME from information_schema.DDL_JOBS where DB_NAME = 'mysql';").Check(testkit.Rows("mysql"))
```

Comment:
I think you should test the table that the user doesn't have the privilege to see it.

Reply:
done